.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,67 @@
+name: zzz [DO NOT RUN] Automated unit tests
+
+on:
```

Comment:
Question for these event hooks: will this not get triggered by the same commit push when the PR is open and you then push to it, as we are not limiting the `pull_request` events? Should we limit this? Not sure.

Something like below would limit runs trigger by the `pull_request` event 🤔 
```yaml
pull_request:
    types: [opened, synchronize]
 ```